src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -3165,25 +3190,38 @@ void AwtWindow::_ModalEnable(void *param)
         AwtWindow::SetModalBlocker(windowHWnd, NULL);
     }
 
-  ret:
+
     env->DeleteGlobalRef(self);
 }
 
 void AwtWindow::_SetOpacity(void* param)
 {
     JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
 
-    OpacityStruct *os = (OpacityStruct *)param;
+    OpacityStruct *os = static_cast<OpacityStruct *>(param);
     jobject self = os->window;
     BYTE iOpacity = (BYTE)os->iOpacity;
 
     PDATA pData;
-    JNI_CHECK_PEER_GOTO(self, ret);
+    if (self == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "self");
+        delete os;
+        return;
+    } else {
+        pData = JNI_GET_PDATA(self);
+        if (pData == NULL) {
+            THROW_NULL_PDATA_IF_NOT_DESTROYED(self);
+            env->DeleteGlobalRef(self);
+            delete os;
+            return;
+        }
+    }
     AwtWindow *window = (AwtWindow *)pData;
 
     window->SetTranslucency(iOpacity, window->isOpaque());
 
-  ret:
+
```

Comment:
I recommend deleting this line, since it does not make sense to have two consecutive blank lines here.